### PR TITLE
Upgrade to KCL 2.0.5

### PR DESCRIPTION
**Pull request by kiambogo**
- Refactor package naming

### CODE REVIEW

File: fs2-aws/src/main/scala/fs2/aws/internal/KinesisProducerClient.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.google.common.util.concurrent.ListenableFuture
+
+trait KinesisProducerClient[F[_]] {
+  def putData(streamName: String, partitionKey: String, data: ByteBuffer)(
+      implicit F: Effect[F]): F[ListenableFuture[UserRecordResult]]
+}
+
+class KinesisProducerClientImpl[F[_]] extends KinesisProducerClient[F] {
+
+  val credentials: AWSCredentialsProviderChain = new DefaultAWSCredentialsProviderChain()
```
Comment (brazir) [inline fs2-aws/src/main/scala/fs2/aws/internal/KinesisProducerClient.scala]: Think from a library perspective you should support different provider chains.  As I imagine this is the one that pulls from the machine and some folks could be using the inline or custom ones.

**Comment by brazir** [comment]
Looks reasonable, but I wonder what the version for the lib should be as you are requiring a new lib as a dependency.

File: fs2-aws/src/main/scala/fs2/aws/internal/KinesisProducerClient.scala (hunk 1 above)
Comment (kiambogo) [reply fs2-aws/src/main/scala/fs2/aws/internal/KinesisProducerClient.scala]: The default provider chain searches through a bunch of different sources (ENV variables, launch arguments, file, and then EC2 metadata) so out of the box provides a bunch of different options for users. By exposing the KinesisProducerClient trait as well we offer users the flexibility to create their own implementation and supply their own credentials. 

https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html

**Comment by kiambogo** [comment]
I think that the versioning should encompass at a minimum a minor version increase. I'm hesitant to cut a major release (1.0.0) as this lib is still heavily under development IMO and 1.0.0 should be reserved for a stable release. In this case, I would propose releasing as 0.27.0.

**Pull request merged**